src/java.compiler/share/classes/javax/lang/model/util/Types.java

```diff
@@ -268,7 +275,10 @@ public interface Types {
      *
      * @param extendsBound  the extends (upper) bound, or {@code null} if none
      * @param superBound    the super (lower) bound, or {@code null} if none
-     * @throws IllegalArgumentException if bounds are not valid
+     *
+     * @throws IllegalArgumentException if bounds are not valid,
+     * including for types that are <em>not</em> {@linkplain ReferenceType
```

Comment:
I find "including for types" hard to parse and connect to the previous clause. Did you mean "e.g. if if the bounds are not reference types" ?

Reply:
Okay; I'll rephrase this to be more clear. Something like:

"throws IllegalArgumentException - if bounds are not valid. Invalid bounds include all types that are not reference types."

The intention of the phrasing is meant to imply there are reference type arguments that could be invalid, per any restrictions from the JLS. Thanks.